docs/siem/detections/api/lists/api-create-list-container.asciidoc

```diff
@@ -0,0 +1,94 @@
+[[lists-api-create-container]]
+=== Create list container
+
+Creates a list container.
+
+A list container groups common <<lists-api-create-list-item, list items>> that
+define exceptions for when detection rule alerts are *not* generated even when
+a rule's other criteria are met.
+
+All list items in the same list container refer to the same type of exception.
+For example, each list item in an `ip` list container excludes an IP address.
+
+==== Request URL
+
+`POST <kibana host>:<port>/api/lists`
+
+==== Request body
+
+A JSON object with these fields:
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`description` |String |Describes the list container. |Yes
+|`id` |String |Unique identifier. |No, automatically created when it is not
+provided.
+|`meta` |Object |Placeholder for metadata about the list container. |No
+|`name` |String |The list container's name. |Yes
+|`type` |String a|Specifies the {es} {ref}/mapping-types.html[data type] of
+excludes the list container holds. Some common examples:
+
+* `keyword`: Many ECS fields are {es} {ref}/keyword.html[keywords]
+* `ip`: IP addresses
+* `ip_range`: {ref}/range.html[Range of IP addresses] (supports IPv4, IPv6, and
+CIDR notation)
+
+|Yes
+
+|==============================================
+
+===== Example requests
+
+Creates a list container for IP addresses:
+
+[source,console]
+--------------------------------------------------
+POST api/lists
+{
+  "id": "internal-ip-excludes",
+  "name": "Exclude internal IP addresses",
+  "description": "Contains list items that exclude internal IP addresses from detection rules.",
+  "type": "ip"
+}
+--------------------------------------------------
+// KIBANA
+
+Creates a list container for a keyword:
+
+[source,console]
+--------------------------------------------------
+POST api/lists
+{
+  "id": "host.name-container",
+  "name": "Exclude hosts",
+  "description": "Contains list items that exclude host names from detection rules.",
+  "type": "keyword"
+}
+--------------------------------------------------
+// KIBANA
+
+
+==== Response code
+
+`200`:: 
+    Indicates a successful call.
+    
+
+==== Response payload
+
+[source,json]
+--------------------------------------------------
+{
+  "id": "internal-ip-excludes",
+  "created_at": "2020-07-07T03:46:39.853Z",
+  "created_by": "Threat Hunter",
+  "description": "Contains list items that exclude internal IP addresses from detection rules.",
+  "name": "Exclude internal IP addresses",
+  "tie_breaker_id": "c670899c-4716-4c33-9135-0ea6c9d8e138",
+  "type": "ip",
+  "updated_at": "2020-07-07T03:46:39.853Z",
+  "updated_by": "Threat Hunter"
+}
```

Comment:
We have had a few minor changes in the last few weeks where you can optionally submit a `_version` with the POST call as well as receive one for OCC.

Here is what the new response looks like when using `post_list.sh`:

```ts
{
  "_version": "WzgzLDFd",
  "id": "ip_list",
  "created_at": "2020-08-04T19:24:51.228Z",
  "created_by": "yo",
  "description": "This list describes bad internet ip",
  "immutable": false,
  "name": "Simple list with an ip",
  "tie_breaker_id": "f42ef979-a6bf-447a-8477-ea34df0197a2",
  "type": "ip",
  "updated_at": "2020-08-04T19:24:51.228Z",
  "updated_by": "yo",
  "version": 1
}
```

The new fields are:

```
version - A numeric number that is only used for seeing when a list changes
immutable - A boolean that tells you if the list is modifiable or not and is for future use, we do nothing with it right now. Users should ignore it and its fine if we don't add it to the documentation at this point
_version - A base 64 encoded value of if_seq_no and if_primary_term used for OCC
```

OCC docs are here:
https://www.elastic.co/guide/en/elasticsearch/reference/current/optimistic-concurrency-control.html

And when the user does an PUT/PATCH (update) of a list they can optionally send in `_version` in those requests and will get back an error/rejection if someone else has modified the list.

Reply:
`_version` is _totally_ optional btw, so if the user does not submit it with a request then they will not have any OCC to worry about if they don't want to. It will just update the same way.